pytorch_lightning/trainer/training_loop.py

```diff
@@ -502,8 +498,8 @@ def run_training_epoch(self):
             self.num_optimizers
         )
 
-        # checkpoint callback
-        self.check_checkpoint_callback(self.should_check_val)
+        # when no val loop is present or fast-dev-run still need to call checkpoints
+        self.check_checkpoint_callback(not (should_check_val and is_overridden('validation_step', model)))
```

Comment:
just for my understanding, this is here in case in the above loop we never run the validation loop (because no val step or not right period). In this case, we still want to run it for the training metric, in case it is monitoring training. You are adding this "not" clause because you also don't want to run it twice. 
Is my reasoning correct?

Reply:
The comment and the condition (`not (should_check_val and is_overridden('validation_step', model))`) were inside of the `check_checkpoint_callback` function, I moved it out because I think it is cleaner to check it outside of the function.

I think your reasoning is correct but @williamFalcon is the one who wrote it originally. Maybe he can confirm it.

Reply:
ok, just be aware that above we have `not A and not B`, which is equivalent to `not (A or B)` but here we have `not (A and B)` which is not the same.

Reply:
Good catch! Messed up de morgan's law lol. I'll fix it